sphinx_gallery/notebook.py

```diff
@@ -125,16 +134,54 @@ def rst2md(text, heading_levels):
     text = re.sub(contents, '', text)
 
     images = re.compile(
-        r'^\.\. image::(.*$)(?:\n *:alt:(.*$)\n)?(?: +:\S+:.*$\n)*',
+        r'^\.\. image::(.*$)((?:\n +:\S+:.*$)*)\n',
         flags=re.M)
+    image_opts = re.compile(r'\n +:(\S+): +(.*)$', flags=re.M)
     text = re.sub(
-        images, lambda match: '![{1}]({0})\n'.format(
-            match.group(1).strip(), (match.group(2) or '').strip()), text)
+        images,
+        lambda match: '<img src="{}"{}>\n'.format(
+            generate_image_src(
+                match.group(1).strip(), gallery_conf, target_dir),
+            re.sub(image_opts, r' \1="\2"', match.group(2) or '')),
+        text)
 
     return text
 
 
-def jupyter_notebook(script_blocks, gallery_conf):
+def generate_image_src(image_path, gallery_conf, target_dir):
+    if re.match(r'https?://', image_path):
+        return image_path
+
+    if not gallery_conf['notebook_images']:
+        return "file://" + image_path.lstrip('/')
+
+    # If absolute path from source directory given
+    if image_path.startswith('/'):
+        # Path should now be relative to source dir, not target dir
+        target_dir = gallery_conf['src_dir']
+        image_path = image_path.lstrip('/')
+    full_path = os.path.join(target_dir, image_path.replace('/', os.sep))
+
+    if isinstance(gallery_conf['notebook_images'], str):
+        # Use as prefix e.g. URL
+        prefix = gallery_conf['notebook_images']
+        rel_path = os.path.relpath(full_path, gallery_conf['src_dir'])
+        return prefix + rel_path.replace(os.sep, '/')
+    else:
+        # True, but not string. Embed as data URI.
+        try:
+            with open(full_path, 'rb') as image_file:
+                data = base64.b64encode(image_file.read())
+        except OSError:
+            warnings.warn(
```

Comment:
Better to raise an `ExtensionError` here? Are there cases where people would actually want this just to warn?

If not going to raise an error, this should be `logger.warning` instead of `warnings.warn`